core/src/toga/documents.py

```diff
@@ -16,13 +17,20 @@
 
 
 class Document(ABC):
-    # Subclasses should override this definition
+    #: A short description of the type of document (e.g., "Text document"). This is a
+    #: class variable that subclasses should define.
+    description: str
 
-    #: A short description of the type of document
-    document_type: str
+    #: A list of extensions that documents of this type might use (e.g.,
+    #: ``["doc", "txt"]``). The list must have at least one extension; the first is the
+    #: default extension for documents of this type. This is a class variable that
+    #: subclasses should define.
+    extensions: list[str]
 
     def __init__(self, app: App):
-        """Create a new Document.
+        """Create a new Document. Do not call this constructor directly - use
+        :meth:`DocumentSet.new`, :meth:`DocumentSet.open` or
+        :meth:`DocumentSet.request_open` instead.
```

Comment:
These links don't work, probably because `:meth:` can't handle names without modules:

```suggestion
        :any:`DocumentSet.new`, :any:`DocumentSet.open` or
        :any:`DocumentSet.request_open` instead.
```